packages/share/ios/Classes/FLTSharePlugin.m

```diff
@@ -91,18 +161,56 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
   }];
 }
 
-+ (void)share:(NSString *)shareText
-           subject:(NSString *)subject
++ (void)share:(NSArray *)shareItems
     withController:(UIViewController *)controller
           atSource:(CGRect)origin {
-  ShareData *data = [[ShareData alloc] initWithSubject:subject text:shareText];
   UIActivityViewController *activityViewController =
-      [[UIActivityViewController alloc] initWithActivityItems:@[ data ] applicationActivities:nil];
+      [[UIActivityViewController alloc] initWithActivityItems:shareItems applicationActivities:nil];
   activityViewController.popoverPresentationController.sourceView = controller.view;
   if (!CGRectIsEmpty(origin)) {
     activityViewController.popoverPresentationController.sourceRect = origin;
   }
   [controller presentViewController:activityViewController animated:YES completion:nil];
 }
 
++ (void)shareText:(NSString *)shareText
+           subject:(NSString *)subject
+    withController:(UIViewController *)controller
+          atSource:(CGRect)origin {
+  ShareData *data = [[ShareData alloc] initWithSubject:subject text:shareText];
+  [self share:@[ data ] withController:controller atSource:origin];
+}
+
++ (void)shareFiles:(NSArray *)paths
+      withMimeType:(NSArray *)mimeTypes
+       withSubject:(NSString *)subject
+          withText:(NSString *)text
+    withController:(UIViewController *)controller
+          atSource:(CGRect)origin {
+  NSMutableArray *items = [[NSMutableArray alloc] init];
+
+  if (text != nil || subject != nil) {
```

Comment:
nits: can be shorten as `if (!text || !subject)`

Reply:
I guess you meant `if (text || subject)` :)